GenderPayGap.WebUI/Controllers/Account/ChangePasswordNewController.cs

```diff
@@ -0,0 +1,115 @@
+using GovUkDesignSystem.Parsers;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GenderPayGap.WebUI.Controllers.Account
+{
+    [Route("manage-account")]
+    public class ChangePasswordNewController : Controller
+    {
+
+        private readonly IDataRepository dataRepository;
+        private readonly IUserRepository userRepository;
+        private readonly EmailSendingService emailSendingService;
+
+        public ChangePasswordNewController(
+            IDataRepository dataRepository,
+            IUserRepository userRepository,
+            EmailSendingService emailSendingService)
+        {
+            this.dataRepository = dataRepository;
+            this.userRepository = userRepository;
+            this.emailSendingService = emailSendingService;
+        }
+        
+        // The user asks to change their password
+        // We direct them to a page that asks for their existing and new passwords
+        [HttpGet("change-password-new")]
+        public IActionResult ChangePasswordGet()
+        {
+            ControllerHelper.ThrowIfAdminIsImpersonatingUser(User);
+
+            // Return the Change Personal Details form
+            return View("ChangePasswordNew", new ChangePasswordNewViewModel());
+        }
+
+        [HttpPost("change-password-new")]
+        [ValidateAntiForgeryToken]
+        [PreventDuplicatePost]
+        public async Task<IActionResult> ChangePasswordPost(ChangePasswordNewViewModel viewModel)
+        {
+            // Check all values are provided and NewPassword is at least 8 characters long
+            viewModel.ParseAndValidateParameters(Request, m => m.CurrentPassword);
+            viewModel.ParseAndValidateParameters(Request, m => m.NewPassword);
+            viewModel.ParseAndValidateParameters(Request, m => m.ConfirmNewPassword);
+            
+            // Get the current user
+            User currentUser = ControllerHelper.GetGpgUserFromAspNetUser(User, dataRepository);
+            
+            // Check that passwords are valid
+            ValidatePasswords(viewModel, currentUser);
+
+            if (viewModel.HasAnyErrors())
+            {
+                return View("ChangePasswordNew", viewModel);
+            }
+            
+            userRepository.UpdateUserPasswordUsingPBKDF2(currentUser, viewModel.NewPassword);
+            
+            // send password change notification
+            emailSendingService.SendChangePasswordCompletedEmail(currentUser.EmailAddress);
+
+            // Set up success notification on Manage Account page
+            string nextPageUrl = Url.Action("ManageAccount", "ManageAccount", new {Area = "Account"});
+            StatusMessageHelper.SetStatusMessage(Response, "Your password has been changed successfully", nextPageUrl);
+
+            // Return user to the Manage Account page
+            return LocalRedirect(nextPageUrl);
+        }
+
+        public async void ValidatePasswords(ChangePasswordNewViewModel viewModel, User currentUser)

```

Comment:
Check how we do this in `AccountCreationController` and `CreateUserAccountViewModel` - we use a `[GpgPasswordValidation]` attribute on the Password field which automatically does our password-strength validation for us - so we don't need to do a lot of this manually in the controller